triedb/pathdb/disklayer.go

```diff
@@ -332,6 +314,7 @@ func (dl *diskLayer) revert(h *history) (*diskLayer, error) {
 	// needs to be reverted is not yet flushed and cached in node
 	// buffer, otherwise, manipulate persistent state directly.
 	if !dl.buffer.empty() {
+		accounts, storages := h.stateSet()
```

Comment:
Do we need to hold the dl.lock when computing the accounts and storages? Or is it better to only compute them if we *have* to revert something because the dl.buffer is not empty?

Reply:
done